kombu/asynchronous/http/curl.py

```diff
@@ -266,7 +266,8 @@ def ioctl(cmd):
             else:
                 setopt(_pycurl.INFILESIZE, len(body))
         elif request.method == 'GET':
-            assert not request.body
+            if request.body:
+                raise AssertionError
```

Comment:
Why is raising `AssertionError` better than using `assert` keyword?

Reply:
Python has an option to compile the optimized bytecode and create the respective .pyo files by using the options `-O` and `-OO`.

During this, these basic optimizations are done:
- All the assert statements are removed
- All docstrings are removed (when `-OO` is selected)
- Value of the debug built-in variable is set to False

Now, if someone never intends to run the optimized bytecode, they can ignore this issue. But, it is still recommended not to have it in non-test files.
A better way for internal self-checks is to explicitly check and raise respective Error using an if statement.

Reply:
I strongly disagree.

If we do decide to make this part of the official contract, we need to raise a proper exception (in this case, a ValueError with an appropriate message).
If we need this during development only, the assert statement should stay and be optimized out of the bytecode in production.

Reply:
Do you want me to revert this change then?

Reply:
Either that or transform them into proper exceptions.